### ISSUE TITLE
Transaction protection - question

### ISSUE THREAD

**Issue by PotOfCoffee2Go** [Belphemur/node-json-db#26]
I am entering final phases of implementing an Auction system using superb node-json-db as the primary database keeping track of guest bidding on items during auction. The Auction repo will be made public in about month from now once stable for forking.

The Problem:
There are two collections that require atomic update; one collection tracks bid requests,  the other collection is the items being bid on. (ie: bid info stored on 'bids' collection is followed immediately by an update of the current price in the 'items' collection.) During tests under load I have discovered inconsistencies between the 'bids' collection and 'items' collection - which is not that surprising given the node event loop. (I could get into some 'Yak shaving' and play around with the ticks - but, as a last resort - I gotta get this project done).

The Question:
Has anyone already implemented atomic update in a fork of  node-json-db?  (awesome!)
If not :(   I will implement - was thinking to use [rwLock](https://github.com/71104/rwlock). But any suggestions or get arounds would be greatly appreciated.

Of course, if I do end up doing it and there is an interested in having transaction protection in node-json-db,  I'll submit a pull request when done .

Thanks
@potofcoffee2go

**Comment by Belphemur** [comment]
Hello @PotOfCoffee2Go,

For a project that relies on multiple/concurrent access + atomicity of the data, I advise against json-db.
Json-DB is more a wrapper to JSON file that's great to manage configuration files or a small set of data.

In your case, you'd better use an RDBMS like Postgresql that would give you more security and better performance than json-DB.

**Comment by PotOfCoffee2Go** [comment]
Hi @Belphemur,

I kinda figured some sane person would recommend to use a  RDBMS.

All of the data for the auction is on a MySql DB running on a AWS EC2 server. It holds all the stuff - the Guest info, login creds, etc. -  as well as Item info, descriptions, images, and stuff. The auction exchange itself only needs around 8 fields - Item ID, current price, status and info on last bid (GuestID, time, bid amount, status).  I already wrote the auction exchange using the MySql DB - establishing a Transaction, doing the writes, the Commit, etc.

The network latency though is right up there - so that is why I gave a shot using node-json-db because I've had great luck with it in the past; and running on OpenShift which has SSD drives - it's faster than fast. I do realize node-json-db was not designed to do this - but... hey, why not give it a shot!

Anyway, just figured out this morning - made a change where am now writing to a single collection with one push( ). Load testing has been running for hours now - no issues! ;)  Not sure how many updates a second it can do as it out performs the number of bids that I can push across the network - how great is that!

So looking forward - I'm going to use it. We'll find out in October - donating my time to great non-profit [Onyx & Breezy](http://onyxandbreezy.org) dedicated to animal welfare.  Their live auction will be on it - so let's keep our fingers crossed. As well as publicly available on Github.

Thank you for your reply!
